projects/packages/my-jetpack/src/products/class-social.php

```diff
@@ -161,40 +168,42 @@ public static function get_status() {
 		return $status;
 	}
 
+	/**
+	 * Get the product-slugs of the paid plans for this product (not including bundles)
+	 *
+	 * @return array
+	 */
+	public static function get_paid_plan_product_slugs() {
+		return array(
+			'jetpack_social_v1_yearly',
+			'jetpack_social_v1_monthly',
+			'jetpack_social_v1_bi_yearly',
+			'jetpack_social_basic_yearly',
+			'jetpack_social_monthly',
+			'jetpack_social_basic_monthly',
+			'jetpack_social_basic_bi_yearly',
+			'jetpack_social_advanced_yearly',
+			'jetpack_social_advanced_monthly',
+			'jetpack_social_advanced_bi_yearly',
+		);
```

Comment:
Should this contain all the legacy products that are no longer used anymore?

Social currently has only those `*_v1_*` products.

Reply:
Hey @manzoorwanijk! 👋😃
> Should this contain all the legacy products that are no longer used anymore?
> Social currently has only those *_v1_* products.

To answer your question: Yes.
Yes, ultimately `has_paid_plan_for_product()` uses this function in question (`get_paid_plan_product_slugs()`) as well as `get_paid_bundles_that_include_product()` to get All the paid plan product_slugs and bundle plan product_slugs that include Social, to ultimately determine if the site has a paid plan for the product (in this case Social).
Some sites/users might still be renewing a legacy Social product that they bought previously, so we need to know if the site/user has _ANY_ Social paid plan (including legacy plans, or even bundled plans that include Social within it). Basically, if the site/user has _Any_ paid plan that includes Social, we want to know about it. 😉

Reply:
Hey

Oh yeah. Thanks.